### PR TITLE
Elastic connector

### PR DESCRIPTION
**Pull request by david-streamlio**
### Motivation

Added a sink connector that writes JSON documents into ElasticSearch

### Modifications

Added new pulsar-io module and associated integration tests

### Result

An ElasticSearch sink connector will be available for use.

### CODE REVIEW

File: tests/integration/src/test/java/org/apache/pulsar/tests/integration/functions/PulsarFunctionsTest.java [hunk 1]
```diff
@@ -60,17 +68,17 @@
...
     @Test
     public void testCassandraSink() throws Exception {
-        testSink(new CassandraSinkTester(), true);
+        testSink(new CassandraSinkTester());
     }
-
-    @Test
-    public void testCassandraArchiveSink() throws Exception {
-        testSink(new CassandraSinkArchiveTester(), false);
+
```
Comment (srkukarni) [inline tests/integration/src/test/java/org/apache/pulsar/tests/integration/functions/PulsarFunctionsTest.java]: Why did we delete the testCassandraArchiveSink test?

File: tests/integration/src/test/java/org/apache/pulsar/tests/integration/functions/PulsarFunctionsTest.java [hunk 2]
```diff
@@ -80,18 +88,23 @@ public void testHdfsSink() throws Exception {
...
+        testSink(new JdbcSinkTester());
+    }
+    
+    @Test
+    public void testElasticSearchSink() throws Exception {
+        testSink(new ElasticSearchSinkTester());
     }
 
-    private void testSink(SinkTester tester, boolean builtin) throws Exception {
+    private void testSink(SinkTester tester) throws Exception {
```
Comment (srkukarni) [inline tests/integration/src/test/java/org/apache/pulsar/tests/integration/functions/PulsarFunctionsTest.java]: I think a bunch of changes were reverted from master by mistake?

**Comment by srkukarni** [comment]
run integration tests

**Comment by david-streamlio** [comment]
How do I view the actual cause of the error..

org.apache.pulsar.tests.integration.docker.ContainerExecException: /pulsar/bin/pulsar-admin sink create --tenant public --namespace default --name test-sink-connector-elastic_search-THREAD-name-exoiydod --sink-type elastic_search --sinkConfig {"elasticSearchUrl":"http://localhost:9200","indexName":"test-index"} --inputs test-sink-connector-**ELASTIC_SEARCH-THREAD-input-topic-midgxwab failed on e463877cd694fe13c88a7b3695cabe3c2840fae0f25b095b1483f2c244ad1d05 with error code 1**
	at org.apache.pulsar.tests.integration.utils.DockerUtils.runCommand(DockerUtils.java:234)
	at org.apache.pulsar.tests.integration.containers.ChaosContainer.execCmd(ChaosContainer.java:109)
	at org.apache.pulsar.tests.integration.functions.PulsarFunctionsTest.submitSinkConnector(PulsarFunctionsTest.java:171)
	at org.apache.pulsar.tests.integration.functions.PulsarFunctionsTest.testSink(PulsarFunctionsTest.java:107)
	at org.apache.pulsar.tests.integration.functions.PulsarFunctionsTest.testElasticSearchSink(PulsarFunctionsTest.java:89)

**Pull request merged**